pkg/controller/noderesource/node_resource_controller.go

```diff
@@ -58,6 +59,9 @@ func (r *NodeResourceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	// get TimeSeriesPrediction result
 	err := r.Client.Get(ctx, req.NamespacedName, tsp)
 	if err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
I think we can use client.IgnoreNotFound(err) too here

Reply:
`r.Recorder.Event(tsp, v1.EventTypeNormal, "FailedGetTimeSeriesPrediction", err.Error())
		klog.Errorf("Failed to get timeSeriesPrediction(%s/%s), %v", req.NamespacedName, req.Name, err)`
		This code create event when get object error,should I delete it?

Reply:
> `r.Recorder.Event(tsp, v1.EventTypeNormal, "FailedGetTimeSeriesPrediction", err.Error()) klog.Errorf("Failed to get timeSeriesPrediction(%s/%s), %v", req.NamespacedName, req.Name, err)` This code create event when get object error,should I delete it?

I have discuss with @kitianFresh , He will refector this controller these days, so I think you can just rebase and keep keep your changes.

Reply:
fixed